single_tech_samples/synapseanalytics/sample3_integration_testing/tests/integrationtests/dataconnectors/adls.py

```diff
@@ -0,0 +1,231 @@
+from io import BytesIO
+from typing import List
+import logging
+import os
+import pytest
+import pandas as pd
+from azure.storage.filedatalake import DataLakeServiceClient
+from dataconnectors import local_file
+
+
+LOG = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="module")
+def adls_account_endpoint():
+    ADLS_ACCOUNT_NAME = os.getenv("ADLS_ACCOUNT_NAME")
+    ADLS_ACCOUNT_URI = os.getenv("ADLS_ACCOUNT_URI")
+    return f"https://{ADLS_ACCOUNT_NAME}.{ADLS_ACCOUNT_URI}/"
+
+
+@pytest.fixture(scope="module")
+def adls_connection_client(azure_credential, adls_account_endpoint):
+    return DataLakeServiceClient(
+        account_url=adls_account_endpoint,
+        credential=azure_credential,
+    )
+
+
+def upload_to_ADLS(
+    adls_connection_client: DataLakeServiceClient,
+    container: str,
+    file_path: str,
+    file_name: str,
+    base_path: str = None,
+    file_contents: bytes = None,
+):
+    """Upload file to ADLS either by reading a local file or directly
+    uploading the file contents
+
+    Args:
+        adls_connection_client (DataLakeServiceClient): ADLS Connection Object
+        container (str): Container Name where file needs to be uploaded
+        file_path (str): File Path of local file to upload
+        file_name (str): File Name of local file to upload
+        base_path (str): Base Folder Path in ADLS where file will be uploaded
+        file_contents (bytes): Contents of the file
```

Comment:
Don't know if this adheres to the right convention, but worth double checking that we are adhering to surfacing to readers what parameters are considered OPTIONAL and what, if any, the default value is

Reply:
Updated!